### PR TITLE
Improve Gamma sampling

### PR DESCRIPTION
**Pull request by adamgayoso**
This fixes:

- a todo to use lax cond in order to speed up a computation during gamma sampling

### CODE REVIEW

File: jax/_src/random.py [hunk 1]
```diff
@@ -957,7 +958,7 @@ def _cond_fn(kXVU):
     true_fun = lambda: lax.ge(lax.log(U), lax.add(lax.mul(X, one_over_two),
                                                       lax.mul(d, lax.add(lax.sub(one, V),
                                                                          lax.log(V)))))
-    false_fun = lambda: False
+    false_fun = lambda: lax.le(one, zero)
```
Comment (jakevdp) [inline jax/_src/random.py:961]: What's the reason for turning a constant into an expression? Mabye `lambda: np.bool_(False)` would be sufficient?

Comment (adamgayoso) [reply jax/_src/random.py]: No good reason. Fixed :)

File: jax/_src/random.py [hunk 2]
```diff
@@ -927,6 +927,7 @@ def _gamma_one(key: KeyArray, alpha, log_space) -> Array:
   # https://en.wikipedia.org/wiki/Gamma_distribution#Generating_gamma-distributed_random_variables
   zero = _lax_const(alpha, 0)
   one = _lax_const(alpha, 1)
+  false = _lax_const(alpha, False)
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply jax/_src/random.py]: Done.

**Review by jakevdp — APPROVED** [review]
Looks good to me, though I'm a bit hesitant to merge this without understanding why `lax.map` was used in the first place, and confirming that that reason no longer is important. Have you been able to discern that?

**Comment by adamgayoso** [comment]
cc @fehiepsi  -- looks like it was added when the grad was still implemented in this file.

I can try a quick benchmark on colab for now?

**Comment by adamgayoso** [comment]
@jakevdp actually it does indeed faster to map on cpu, I had a bug in my benchmark. Also would probably good to leave these potential changes in separate PRs. Therefore I changed this PR to only include the lax cond change

File: jax/_src/random.py [hunk 3]
```diff
@@ -953,13 +953,12 @@ def _gamma_one(key: KeyArray, alpha, log_space) -> Array:
...
-    # if the first condition is satisfied
-    cond = lax.bitwise_and(lax.ge(U, lax.sub(one, lax.mul(squeeze_const, lax.mul(X, X)))),
-                           lax.ge(lax.log(U), lax.add(lax.mul(X, one_over_two),
+    pred = lax.ge(U, lax.sub(one, lax.mul(squeeze_const, lax.mul(X, X))))
+    true_fun = lambda: lax.ge(lax.log(U), lax.add(lax.mul(X, one_over_two),
                                                       lax.mul(d, lax.add(lax.sub(one, V),
-                                                                         lax.log(V))))))
+                                                                         lax.log(V)))))
+    false_fun = lambda: np.bool_(False)
+    cond = lax.cond(pred, true_fun, false_fun)
```
Comment (fehiepsi) [inline jax/_src/random.py:961]: Does this change make things faster? Could you post some benchmark numbers?

**Comment by jakevdp** [comment]
It seems like this has become stale. I'm going to close for now – feel free to re-open if you'd like to continue working on this!

**Pull request closed without merging**